tests/acceptance/stepDefinitions/publicLinkContext.js

```diff
@@ -5,48 +5,139 @@ const sharingHelper = require('../helpers/sharingHelper')
+Then('user {string} should not have any public link',
+  async function (sharer) {
+    const resp = await sharingHelper.getAllShares(sharer)
+    assert.deepStrictEqual(
+      resp, [], 'User has no empty shares. Response: ' + resp)
+  })
 
+Then('the fields of the last public link share response of user {string} should include',
+  function (linkCreator, dataTable) {
+    const fieldsData = dataTable.rowsHash()
+    return sharingHelper.assertUserLastPublicShareDetails(linkCreator, fieldsData)
+  })
+
+When('the public uses the webUI to access the last public link created by user {string} with password {string}', async function (linkCreator, password) {
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShareToken = lastShare.token
+  await client.page.publicLinkFilesPage().navigateAndWaitForPasswordPage(lastShareToken)
   return client.page.publicLinkPasswordPage().submitPublicLinkPassword(password)
 })
 
+Then('the user should see an error message on the public link share dialog saying {string}',
+  function (errorMessage) {
+    return client.page.FilesPageElement
+      .publicLinksDialog()
+      .assertErrorMessageForPublicLinks(errorMessage)
+  })
+
+Then('the public link of the user {string} should not have been generated', async function (sharer) {
+  const resp = await sharingHelper.fetchLastPublicLinkShare(sharer)
+  assert.strictEqual(
+    resp, undefined, 'Error with public link shares! ' + resp
+  )
+})
+
+Then('the public should not get access to the publicly shared file', function () {
+  return client
+    .page
+    .publicLinkPasswordPage()
+    .assertResourceAccessDenied()
+})
+
+When('the user tries to edit the public link named {string} of file/folder/resource {string} changing following',
+  async function (linkName, resource, dataTable) {
+    const editData = dataTable.rowsHash()
+    await client.page.FilesPageElement
+      .filesList()
+      .closeSidebar(100)
+      .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
+      .editPublicLink(linkName, editData, false)
+  })
+
+When('the user edits the public link named {string} of file/folder/resource {string} changing following',
+  async function (linkName, resource, dataTable) {
+    const editData = dataTable.rowsHash()
+    await client.page.FilesPageElement
+      .filesList()
+      .closeSidebar(100)
+      .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
+      .editPublicLink(linkName, editData, true)
+  })
+
+When('the user tries to edit expiration of the public link named {string} of file {string} to past date {string}',
+  async function (linkName, resource, pastDate) {
+    await client.page.FilesPageElement
+      .filesList()
+      .closeSidebar(100)
+      .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
+      .assertDisabledPastExpiryDate(linkName, pastDate)
```

Comment:
Does this need to be a special step? Can we have a step similar to `assertDateIsDisabled(linkName, date)`?

Reply:
:rocket: